lib/elixir/src/elixir_errors.erl

```diff
@@ -55,6 +55,13 @@ parse_error(Line, File, Error, <<>>) ->
 parse_error(Line, File, <<"syntax error before: ">>, <<"'end'">>) ->
   do_raise(Line, File, 'Elixir.SyntaxError', <<"unexpected token: end">>);
 
+%% Produce a human-readable message for errors before a sigil
+parse_error(Line, File, <<"syntax error before: ">>, <<"{sigil,", _Rest/binary>> = Full) ->
+  {ok, Tokens, _} = erl_scan:string(binary_to_list(Full)),
+  {ok, {sigil, _, _, [Content], _}} = erl_parse:parse_term(Tokens ++ [{dot, 1}]),
```

Comment:
I think the sigil marker is the third element of the tuple. Maybe it is worth including it in the error message? Something like `sigil ~s with content ...`?